pylint/test/test_self.py

```diff
@@ -458,3 +458,23 @@ def test_getdefaultencoding_crashes_with_lc_ctype_utf8(self):
         module = join(HERE, 'regrtest_data', 'application_crash.py')
         with _configure_lc_ctype('UTF-8'):
             self._test_output([module, '-E'], expected_output=expected_output)
+
+    @pytest.mark.skipif(sys.platform == 'win32', reason='only occurs on *nix')
+    def test_parseable_file_path(self):
```

Comment:
We could probably already create this directory in `pylint/test/regrtest_data`. Then in the test, we only have to call `self._test_output`.

Reply:
@PCManticore thanks for the input, however I think pre-create directory in `pylint/test/regtest_data` is not able to reproduce this issue, as `reporter.path_strip_prefix` is the absolute path all the way from `/` to current folder.

Take Travis CI with Python 3.6 env as example, the `reporter.path_strip_prefix` is actually `/home/travis/build/PyCQA/pylint/.tox/py36/lib/python3.6/site-packages/pylint/test/regrtest_data/package/`. This path can be changed with different python version and different run time environment, that's why I decided to dynamically create it based on current environment.

Reply:
Agreed, sounds like this is the best way to test this scenario.